### PR TITLE
ISPN-10839 Configuration Doc Updates

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.redhat.com/browse/ISPN-10839

Needs backport to 10.1. See the Jira for details and review draft.

### CODE REVIEW

File: documentation/src/main/asciidoc/topics/ref_config_programmatic.adoc [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+Create new Configuration objects with the `ConfigurationBuilder` class and then
+define cache configurations with the Cache Manager.
+
+[NOTE]
+====
+The examples in this section use `EmbeddedCacheManager`, which is a Cache
+Manager that runs in the same JVM as the client.
+
+To configure caches remotely with HotRod clients, you use `RemoteCacheManager`.
+Refer to the HotRod documentation for more information.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/ref_config_programmatic.adoc]: Can we link to the HotRod documentation? Or does that not work with how things are separated?

File: documentation/src/main/asciidoc/topics/ref_config_declarative.adoc [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+<5> names a cache instance that uses invalidation mode.
+<6> names a cache instance that uses replicated mode.
+<7> names a cache instance that uses distributed mode.
+
+.Default caches
+{brandname} does not provide an implicit default cache. You must specify which
+named cache is the default.
+
+Additional named caches inherit settings from the default cache. You can,
+however, override or extend configuration in each named cache.
```
Comment (ryanemerson) [inline documentation/src/main/asciidoc/topics/ref_config_declarative.adoc]: I think this needs clarifying to make it clear that "additional named caches" refers to when a user calls getCache("cache-name"). It does not mean that another declaritevely named cache will inherit it's configuration from the default cache.

This is made somewhat more confusing by the fact we also have to explain <cache-type>-configuration and how multiple caches can use it via:

<distributed-cache name="repl" configuration="configuration-name"/>

Comment (ryanemerson) [reply documentation/src/main/asciidoc/topics/ref_config_declarative.adoc]: @tristantarrant I can't keep up, do calls to `getCache("new name")` create a new cache using the default config now?

File: documentation/src/main/asciidoc/topics/ref_config_programmatic.adoc (hunk 1 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/ref_config_programmatic.adoc]: @ryanemerson we do link to the HotRod docs but from the `.References` section. Linking between guides in community is really straightforward. I've added attributes so all you need to do is `link:{hotrod_docs}[HotRod Java Clients]`. 

you can add anchors as well like `link:{hotrod_docs}#anchor_name[HotRod Java Clients]`.

also pushing links to a reference section keeps readers "on task". inline linking can lead to distractions. I could go all day with this... let me know what you think though. I can show you some examples and explain in more detail.

File: documentation/src/main/asciidoc/topics/ref_config_declarative.adoc (hunk 2 above)
Comment (oraNod) [reply documentation/src/main/asciidoc/topics/ref_config_declarative.adoc]: @ryanemerson thanks for clarifying that detail. tbh I wasn't sure about it and also felt it wasn't clear. I'm going to remove that sentence as it doesn't belong in the declarative config content.

Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/ref_config_declarative.adoc]: Caches no longer implicitly inherit from the default cache.

Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/ref_config_declarative.adoc]: The default cache is just any named cache that you obtain when you invoke `getCache()`

**Comment by ryanemerson** [comment]
Thanks @oraNod

**Pull request merged**